Hacktoberfest [JENKINS-69731] Update shared-libraries.adoc (#5537)
jenkins-infra/jenkins.io: content/doc/book/pipeline/shared-libraries.adoc

video::eV7roTXrEqg[youtube,width=800,height=420]

=== Statically loading evaluated versions of libraries

> As of currently unreleased version of the _Pipeline: Shared Groovy Libraries_ plugin,
> with features proposed in
> link:https://github.com/jenkinsci/pipeline-groovy-lib-plugin/pull/19[PR #19]
> (link:https://issues.jenkins.io/browse/JENKINS-69731[issue JENKINS-69731])

Reviewer: Thanks for phrasing it that way to make it clear to reviewers like me that the feature is not yet released.  Very nice technique!
```suggestion
```

Author: I suppose you did not mean removing these lines, just yet? :)

(If this PR is indeed only merged after the feature release, this should get rephrased similar to `library` step description)

there are new options for the `@Library` annotation with limited run-time evaluation
of the version specified for the library. In order to use these features, you should
explicitly allow them in library configuration (e.g. Global Jenkins configuration of
your instance).

While the markup for these features is intentionally similar to what you would write
for Groovy code to simplify maintenance, keep in mind that the version tokens are
handled by the plugin before it would load libraries and compile the pipeline script.

One addressed use-case regards loading of "same version of library as the version of
Pipeline script from SCM" by specifying literally +`${BRANCH_NAME}`+ as the library
version in the pipeline script, e.g.:

[source,groovy]
----
@Library('my-shared-library@${BRANCH_NAME}') _
----

> The plugin would try several methods to determine the branch name to use, check if
> such "version" is accepted by SCM source of the library, and fall back to default
> version if not.

Reviewer: Simpler verbs and a few minor changes for your consideration
```suggestion
The plugin tries several methods to determine the branch name to use, checks if
the "version" is accepted by the SCM source of the library, and falls back to the default
version if not.
```

This was verified to work for:
* MultiBranch Project pipelines both with "Modern SCM" and with "Single repository & branch"
  (in case of the latter, the "name" you specify for the source is what matters for MBP more
  than the actual SCM branch);
* Pipelines backed by a single SCM source with a "Branch Specifier" like `*/main`;
* "In-line" Pipelines not backed by an SCM (just fall back to default version).

> Another addressed use-case regards loading of some version of library specified by

Reviewer: Possibly simpler phrasing?
```suggestion
The plugin can also load a library version specified by
```

Author: Not sure about this change (alone) since an earlier paragraph stressed that "one use-case is..." so here "another"...

an environment variable value defined when the job starts (e.g. coming from Global

> Jenkins configuration, from build agent configuration, or injected by some plugins):

Reviewer: ```suggestion
Jenkins configuration, from build agent configuration, or injected by a plugin):
```
